wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/linking/expression/files/EngineFunctionEvaluators.scala

```diff
@@ -13,17 +13,19 @@ import wdl.model.draft3.graph.expression.FileEvaluator.ops._
 import wom.expression.IoFunctionSet
 import wom.types._
 import wom.values._
+import wdl.draft3.transforms.wdlom2wdl.WdlWriter.ops._
+import wdl.draft3.transforms.wdlom2wdl.WdlWriterImpl.expressionElementWriter
 
 object EngineFunctionEvaluators {
 
   private def evaluateToFile(forFunction: String, a: ExpressionElement, inputs: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[Set[WomFile]] = {
     a match {
-      case _: IdentifierLookup | _: IdentifierMemberAccess => Set.empty[WomFile].validNel
+      case _: IdentifierLookup | _: IdentifierMemberAccess | _: IndexAccess => Set.empty[WomFile].validNel
```

Comment:
This is the extra case which fixes the bug. Everything else is context or test cases